### ISSUE TITLE
Unfeaturized entities for specified intents

### ISSUE THREAD

**Issue by akelad** [RasaHQ/rasa_core#726]
In the domain file you should be able to specify whether an entity should be ignored for certain intents (i.e. the next action predicted should not change dependent on whether this entity is present or not)

Aditionally you should be able to specify whether or not automatic slot filling should happen for this intent

**Comment by Ghostvv** [comment]
for `this` intent? or for other intents?

**Comment by akelad** [comment]
well i think we want to be able to turn it off for this intent

**Comment by tmbo** [comment]
There is something else we should try: How about not using entities as features at all but only slots (and no other change).

This will ignore entities that are not relevant to the intent completely. If an entity is relevant for another intent and there is a slot to store its value, it will be relevant for all intents. But because of the story concatenation, the slot will sometimes be set and sometimes not, introducing some training noise we don't have for entities. So I think in this case, the effect of an entity being present shouldn't throw of the model to much. What do you think?

**Comment by Ghostvv** [comment]
if we take our generalization example, `* correct{"location": "paris"}` entity is the key here, and slot was set before (but we use unfeaturized slots)

**Comment by fdevillard** [comment]
In the PR #763 the idea is to add which entities are ignored. What about the opposite: which entities should be considered in the policy's features? And, if nothing is specified, then we come back on the current behavior (i.e. considering all entities) for backward compatibility?

From my point of view, the case where the entities change the policy's prediction are more the exception than the rule. This therefore suggests that it might be better to specify which entities are relevant in the dialogue than which ones are not.

**Comment by akelad** [comment]
So the idea of this PR is to specify a list of intents for which entities should be ignored. this is for the case where you don't care which entities are picked up for a certain intent as it's irrelevant to the flow of the dialogue.
I'm not sure I agree that entities changing the policy's prediction are more the exception than the rule. e.g. if with an insurance bot you could have the user say something like "I have an insurance claim" or "I have an insurance claim, my insurance number is 123" - you'd want to handle these differently

**Comment by fdevillard** [comment]
Yep, ok. I'm totally agree that the featurization of the entities' is important, as well as the unfeaturization. I just wanted to point out that, in my case, entities play a minor role on the policy. 

My use cases are generally the following: if the user has filled some slots, the API call will be more _precise_. But, then, the result of this API call changes the dialogue (and not the entities). That's why, from my point of view, the influence of entities in the dialogue is more the exception than the rule.

But anyway, thanks for this issue/PR :)

**Comment by akelad** [comment]
Cool, thanks for your input! this PR should hopefully work for your use case too :)
